src/contrib/inference/dynamichmc.jl

```diff
@@ -55,44 +64,41 @@ function DynamicPPL.initialstep(
     model::Model,
     spl::Sampler{<:DynamicNUTS},
     vi::AbstractVarInfo;
-    N::Int,
     kwargs...
 )
-    # Set up lp function.
-    function _lp(x)
-        gradient_logp(x, vi, model, spl)
-    end
-
-    link!(vi, spl)
-    l, dl = _lp(vi[spl])
-    while !isfinite(l) || !isfinite(dl)
-        model(vi, SampleFromUniform())
-        link!(vi, spl)
-        l, dl = _lp(vi[spl])
-    end
-
-    if spl.selector.tag == :default && !islinked(vi, spl)
-        link!(vi, spl)
-        model(vi, spl)
+    # Ensure that initial sample is in unconstrained space.
+    if !DynamicPPL.islinked(vi, spl)
```

Comment:
AFAIK `link!` won't actually do anything if the variables are already transformed.

`islinked` and `link!` both perform the same check:
- `link!`: https://github.com/TuringLang/DynamicPPL.jl/blob/6ac3922bf09cf8e0d124b448353dd18b7afee00d/src/varinfo.jl#L701
- `islinked`: https://github.com/TuringLang/DynamicPPL.jl/blob/6ac3922bf09cf8e0d124b448353dd18b7afee00d/src/varinfo.jl#L797

I.e. `islinked` is redundant.

EDIT: Nvm, I see that there's a warning beeing logged if we're linking an already linked `vi` :+1: